Source/ablastr/mesh_refinement/coarsen.H

```diff
@@ -1,25 +1,35 @@
-#ifndef WARPX_COARSEN_MR_H_
-#define WARPX_COARSEN_MR_H_
+/* Copyright 2022 Edoardo Zoni, Remi Lehe, Prabhat Kumar, Axel Huebl
+ *
+ * This file is part of ABLASTR.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef ABLASTR_MR_COARSEN_H_
+#define ABLASTR_MR_COARSEN_H_
+
 
 #include <AMReX_Array.H>
 #include <AMReX_Array4.H>
+#include <AMReX_BLassert.H>
 #include <AMReX_Extension.H>
 #include <AMReX_GpuQualifiers.H>
+#include <AMReX_Math.H>
 #include <AMReX_REAL.H>
 
 #include <AMReX_BaseFwd.H>
 
 #include <cstdlib>
 
-namespace CoarsenMR{
-
-    using namespace amrex;
 
+namespace ablastr::mesh_refinement
+{
     /**
      * \brief Interpolates the floating point data contained in the source Array4
      *        \c arr_src, extracted from a fine MultiFab, with weights defined in
      *        such a way that the total charge is preserved.
      *
+     * The input (sf) and output (sc) staggering need to be the same.
```

Comment:
@EZoni let us clarify this and the difference to the other function.

Reply:
I think I agree with this. Also looking at my notes, I think the interpolation operation (fine to coarse, or coarsening indeed) used for mesh refinement purposes (not I/O) was derived and implemented only for the case where the fine and the coarse grids have the same index type. Which, I think, represents a very common and reasonable scenario.

Reply:
Thanks, perfect.